jupyterhub/objects.py

```diff
@@ -174,7 +174,7 @@ def wait_up(self, timeout=10, http=False, ssl_context=None):
             )
 
             return wait_for_http_server(
-                self.url, timeout=timeout, ssl_context=ssl_context
+                urljoin(self.url, "api"), timeout=timeout, ssl_context=ssl_context
```

Comment:
This `/api/` bit is specific to waiting for user servers, which the Server is for any http server, so I think either `wait_up` needs to add a path argument or the Server on which `wait_up` is called needs to be constructed via `Server.from_url(urljoin(server.url, "api")` to add it to `self.url`.